R/dbplyr.R

```diff
@@ -0,0 +1,274 @@
+db_insert_into.Pool <- function(con, table, values, ...) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  db_insert_into(db_con, table = table, values = values, ... = ...)
+}
+
+#' @export
+db_list_tables.Pool <- function(con) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  db_list_tables(db_con)
+}
+
+#' @export
+db_query_fields.Pool <- function(con, sql, ...) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  db_query_fields(db_con, sql = sql, ... = ...)
+}
+
+#' @export
+db_query_rows.Pool <- function(con, sql, ...) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  db_query_rows(db_con, sql = sql, ... = ...)
+}
+
+#' @export
+db_rollback.Pool <- function(con, ...) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  db_rollback(db_con, ... = ...)
+}
+
+#' @export
+db_save_query.Pool <- function(con, sql, name, temporary = TRUE, ...) {
+  stopIfTemporary(temporary)
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  db_save_query(db_con, sql = sql, name = name,
+    temporary = temporary, ... = ...)
+}
+
+#' @export
+db_write_table.Pool <- function(con, table, types, values,
+  temporary = FALSE, ...) {
+    stopIfTemporary(temporary)
+    db_con <- poolCheckout(con)
+    on.exit(poolReturn(db_con))
+    db_write_table(db_con, table = table, types = types,
+      values = values, temporary = temporary, ... = ...)
+}
+
+#' @export
+sql_escape_ident.Pool <- function(con, x) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  sql_escape_ident(db_con, x = x)
+}
+
+#' @export
+sql_escape_string.Pool <- function(con, x) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  sql_escape_string(db_con, x = x)
+}
+
+#' @export
+sql_join.Pool <- function(con, x, y, vars, type = "inner", by = NULL, ...) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  sql_join(db_con, x = x, y = y, vars = vars, type = type,
+    by = by, ... = ...)
+}
+
+#' @export
+sql_select.Pool <- function(con, select, from, where = NULL,
+  group_by = NULL, having = NULL, order_by = NULL, limit = NULL,
+  distinct = FALSE, ...) {
+    db_con <- poolCheckout(con)
+    on.exit(poolReturn(db_con))
+    sql_select(db_con, select = select, from = from,
+      where = where, group_by = group_by, having = having,
+      order_by = order_by, limit = limit, distinct = distinct, ... = ...)
+}
+
+#' @export
+sql_semi_join.Pool <- function(con, x, y, anti = FALSE, by = NULL, ...) {
+  db_con <- poolCheckout(con)
+  on.exit(poolReturn(db_con))
+  sql_semi_join(db_con, x = x, y = y, anti = anti, by = by,
+    ... = ...)
+}
+
+random_table_name <- function(n = 10) {
+  paste0(sample(letters, n, replace = TRUE), collapse = "")
+}
+
+#' @export
+sql_subquery.Pool <- function(con, from,
+  name = random_table_name(), ...) {
+    db_con <- poolCheckout(con)
+    on.exit(poolReturn(db_con))
+    sql_subquery(db_con, from = from, name = name, ... = ...)
+}
```

Comment:
another worry I have is if this function is amenable to `pool`. Is the `name = random_table_name()`'s value actually made into a table using `create_table` or something?